### ISSUE TITLE
letsencrypt manual authentication internal server error - Error creating new authz

### ISSUE THREAD

**Issue by centminmod** [letsencrypt/letsencrypt#735]
This was following on from https://github.com/letsencrypt/letsencrypt/issues/734 

with CentOS 6.7 with Python 2.7 installed via IUS Community YUM repo for CentOS 6 https://iuscommunity.org/pages/Repos.html installed via

    yum -y install python27 python27-devel python27-pip python27-setuptools python27-tools python27-virtualenv --enablerepo=ius

running commands

     virtualenv --no-site-packages -p python2.7 venv
    ./venv/bin/pip install -r requirements.txt acme/ . letsencrypt-apache/ letsencrypt-nginx/
    ./venv/bin/letsencrypt -d le1.http2ssl.xyz auth

get

    An unexpected error occurred. Please see the logfiles in /var/log/letsencrypt for more details

and log at /var/log/letsencrypt/letsencrypt.log  says

    2015-08-29 05:40:38,204:DEBUG:acme.jose.json_util:Omitted empty fields: challenges=None, combinations=None, status=None, expires=None
    2015-08-29 05:40:38,204:DEBUG:acme.client:Serialized JSON: {"identifier": {"type": "dns", "value": "le1.http2ssl.xyz"}, "resource": "new-authz"}
    2015-08-29 05:40:38,205:DEBUG:acme.jose.json_util:Omitted empty fields: alg=None, x5c=(), crit=(), kid=None, jwk=None, typ=None, jku=None, x5tS256=None, cty=None, x5u=None, x5t=None
    2015-08-29 05:40:38,209:DEBUG:acme.jose.json_util:Omitted empty fields: x5c=(), crit=(), kid=None, typ=None, nonce=None, jku=None, x5tS256=None, cty=None, x5u=None, x5t=None
    2015-08-29 05:40:38,209:DEBUG:root:Sending POST request to https://acme-staging.api.letsencrypt.org/acme/new-authz. args: (), kwargs: {'data': '{"header": {"alg": "RS256", "jwk": {"e": "AQAB", "kty": "RSA", "n": "mw6R4otmOQy9jot4gHd7MvS4TkbCO0U4OswY9PleU6FA_Xu694ReC3zLdrJhjqb4dEs7BQbeR1CGT0W3z69Ngv7hP8T_UKQosL9dlUr2JMechC2WK_q7UmoV5TZjnqE1R6yhbGcLQbEI_QUvsA7hy4fCO7drzu6GrGeZCe1rszmb4xJ4EvcRVT_NEZBXJ6a07PC_jLBn_TNFlsLuXbtOjoQuXX4caahN_rWvHphNqzjbqSR_yztETVAkbYcCkjTYwyfx1vU1TiymVu60uQno317ROTrVj2QE5MsErh_nNRZ8n8HwvHtwH4D2oB7tzX6oaf_3sX0_dh5fFh4nL4BTwQ"}}, "protected": "eyJub25jZSI6ICJIc1ctVzdOWWxscVFzZmRNX1VJR1piUWpwSXY5TWNaQVkyTkI3ZktQRnQ0In0", "payload": "eyJpZGVudGlmaWVyIjogeyJ0eXBlIjogImRucyIsICJ2YWx1ZSI6ICJsZTEuaHR0cDJzc2wueHl6In0sICJyZXNvdXJjZSI6ICJuZXctYXV0aHoifQ", "signature": "Bk72E7-UZqCXRSbLiDH_XsetEjky6FBic9p5Bno5KEAaHkvD_4KwEfihBRBDsXB8HoQifeoQv0c3Mqr-8Jd-gwkNTxG1W0yWMqxCo7qrDSdsdyAmX7N4nkdKBk3X8UWuuvFrjsTlZacw7p9eCg_uDUdGXMbzcOJ8JOt3H0LpY85YU4cyXQ9TZTpL6vP-    M7SOOAvj7EuVTlrVC6km6vcS7DLMD6zG9vcXrUB4xTevcNavC5C8U5NGs9kAjhWlRMmh2qCdUP9LBjvbW8--4s9c-00diaU8ltlkn52jXphmTjIaxAhJiRqt1Jng8TAN9Kp9mB-IU2ysCkT4oA7kS9fENA"}'}
    2015-08-29 05:40:38,210:INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): acme-staging.api.letsencrypt.org
    2015-08-29 05:40:38,898:DEBUG:requests.packages.urllib3.connectionpool:"POST /acme/new-authz HTTP/1.1" 500 76
    2015-08-29 05:40:38,900:DEBUG:root:Received <Response [500]>. Headers: {'content-length': '76', 'expires': 'Sat, 29 Aug 2015 05:40:38 GMT', 'server': 'nginx', 'connection': 'close', 'pragma': 'no-cache', 'cache-control': 'max-age=0, no-cache, no-store', 'date': 'Sat, 29 Aug 2015 05:40:38 GMT', 'access-control-allow-origin': '*', 'content-type': 'application/problem+json', 'replay-nonce':     'fTMMT4wGK5Fq2gCKSpQ4yzlDAcRWwSQyrrRaKzUSywM'}. Content: '{"type":"urn:acme:error:serverInternal","detail":"Error creating new authz"}'
    2015-08-29 05:40:38,901:DEBUG:acme.client:Storing nonce: '}3\x0cO\x8c\x06+\x91j\xda\x00\x8aJ\x948\xcb9C\x01\xc4V\xc1$2\xae\xb4Z+5\x12\xcb\x03'
    2015-08-29 05:40:38,901:DEBUG:acme.client:Received response <Response [500]> (headers: {'content-length': '76', 'expires': 'Sat, 29 Aug 2015 05:40:38 GMT', 'server': 'nginx', 'connection': 'close', 'pragma': 'no-cache', 'cache-control': 'max-age=0, no-cache, no-store', 'date': 'Sat, 29 Aug 2015 05:40:38 GMT', 'access-control-allow-origin': '*', 'content-type': 'application/problem+json', 'replay-nonce':     'fTMMT4wGK5Fq2gCKSpQ4yzlDAcRWwSQyrrRaKzUSywM'}): '{"type":"urn:acme:error:serverInternal","detail":"Error creating new authz"}'
    2015-08-29 05:40:38,903:DEBUG:letsencrypt.cli:Exiting abnormally:
    Traceback (most recent call last):
      File "./venv/bin/letsencrypt", line 9, in <module>
        load_entry_point('letsencrypt==0.1', 'console_scripts', 'letsencrypt')()
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/letsencrypt/cli.py", line 778, in main
        return args.func(args, config, plugins)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/letsencrypt/cli.py", line 225, in auth
        domains, authenticator, installer, plugins):
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/letsencrypt/client.py", line 240, in obtain_and_enroll_certificate
        certr, chain, key, _ = self.obtain_certificate(domains)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/letsencrypt/client.py", line 216, in obtain_certificate
        return self._obtain_certificate(domains, csr) + (key, csr)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/letsencrypt/client.py", line 174, in _obtain_certificate
        authzr = self.auth_handler.get_authorizations(domains)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/letsencrypt/auth_handler.py", line 73, in get_authorizations
        domain, self.account.regr.new_authzr_uri)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/acme/client.py", line 202, in request_domain_challenges
        typ=messages.IDENTIFIER_FQDN, value=domain), new_authz_uri)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/acme/client.py", line 182, in request_challenges
        response = self.net.post(new_authzr_uri, new_authz)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/acme/client.py", line 596, in post
        return self._check_response(response, content_type=content_type)
      File "/root/tools/letsencrypt/venv/lib/python2.7/site-packages/acme/client.py", line 514, in _check_response
        raise messages.Error.from_json(jobj)
    Error: serverInternal :: The server experienced an internal error :: Error creating new authz

**Comment by kuba** [comment]
Could you please try if standalone plugin works?

If not, please remove (with prior backup!) the following directories:
- `/etc/letsencrypt`
- `/var/lib/letsencrypt`

and retry. Does it help?

**Comment by centminmod** [comment]
if i have a working python 2.6 setup in same folder with ssl certs obtained, should i be removing those directories ?

what's best way to test both python 2.6 and 2.7 and still retain the ssl certs ?

**Comment by kuba** [comment]
No, you don't have to remove them just because you're using both 2.6 and 2.7 at the same time. I suggested this solution because of #727, which could indicate that Boulder team has recently wiped the accounts database, which is often exhibited in client through the cryptic messages you're currently getting.

**Comment by centminmod** [comment]
so how do I run python 2.7 version if i did this for python 2.6 version

    mkdir -p /root/tools
    cd /root/tools
    git clone https://github.com/letsencrypt/letsencrypt
    cd letsencrypt
    ./bootstrap/centos.sh
    virtualenv --no-site-packages -p python2 venv
    ./venv/bin/pip install -r requirements.txt acme/ . letsencrypt-apache/ letsencrypt-nginx/
    ./venv/bin/letsencrypt -d le1.http2ssl.xyz auth

**Comment by kuba** [comment]
Did you fix your original problem?

You will have to set up separate virtualenvs for each Python version:
```
virtualenv --no-site-packages -p python2.6 venv2.6
virtualenv --no-site-packages -p python2.7 venv2.7
./venv2.6/bin/pip install -r requirements.txt ...
./venv2.7/bin/pip install -r requirements.txt ...
./venv2.6/bin/letsencrypt
./venv2.7/bin/letsencrypt
```

**Comment by centminmod** [comment]
@kuba yup fixed now seems to work with python 2.7 and manual authenticator mode

    Press ENTER to continue
    /root/tools/letsencrypt/venv2.7/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:768: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is     strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
      InsecureRequestWarning)
    
    IMPORTANT NOTES:
     - Automatic renewal and deployment has been enabled for your
       certificate. These settings can be configured in the directories
       under /etc/letsencrypt/configs.
    
    ls -lAhrt /etc/letsencrypt/live/le2.http2ssl.xyz/
    total 0
    lrwxrwxrwx 1 root root 43 Aug 30 08:07 privkey.pem -> ../../archive/le2.http2ssl.xyz/privkey1.pem
    lrwxrwxrwx 1 root root 45 Aug 30 08:07 fullchain.pem -> ../../archive/le2.http2ssl.xyz/fullchain1.pem
    lrwxrwxrwx 1 root root 41 Aug 30 08:07 chain.pem -> ../../archive/le2.http2ssl.xyz/chain1.pem
    lrwxrwxrwx 1 root root 40 Aug 30 08:07 cert.pem -> ../../archive/le2.http2ssl.xyz/cert1.pem

**Comment by kuba** [comment]
If it's fixed, please close the issue.
